src/js/tabs/firmware_flasher.js

```diff
@@ -379,17 +379,24 @@ TABS.firmware_flasher.initialize = function (callback) {
                         analytics.setFirmwareData(analytics.DATA.FIRMWARE_NAME, file.name);
                         var reader = new FileReader();
 
+                        // Max size in bytes of the local file to load and flash
+                        const MAX_FLASH_LOCAL_FILE_SIZE = 2*1024*1024;
```

Comment:
How is this limit defined?

Also, not sure we need to check for a limit here anyway, what problems do we expect to have if a file is larger than a set limit at this stage?

Reply:
I don't know. Maybe is a security measure if the user selects a wrong file? I can remove the limit if you want.

Reply:
Yes please. Worst that can happen is that configurator crashes. This will happen as well if they load a 'CLI `diff`' of the same size, so no reason to be overly cautious here.
Also, 2 MB seems to be an arbitrary limit - since there is a lot of flexibility in how a .hex file can be organised, the size of the file can vary a lot even if the data that is actually flashed is the same.

Reply:
The 1MB limit was added back in baseflight-configurator. As someone who occasionally clicks on the wrong thing, a limit does sound like a good idea. I think the limit should exceed what we'd consider reasonable. My thought would be limit it to 64 MB, Just to save someone having to wait for the configurator to crash if they accidentally try and load a .iso

Reply:
Not really sure we should be policing this - there are lots of other ways that users can do 'bad things' with configurator (or pretty much any other app that does file I/O). (E.g. they can use 'Save Output' to overwrite files that are part of the configurator installation, or the operating system.) Making a patchwork effort to fix some of these cases (others are unsolvable) will not solve the problem, and just add more code that then can break, like in the case at hand.